src/dune_engine/sandbox_mode.ml

```diff
@@ -109,7 +109,10 @@ end
 
 (* these should be listed in the default order of preference *)
 let all_except_patch_back_source_tree =
-  [ None; Some Symlink; Some Copy; Some Hardlink ]
+  if Sys.win32 then
+    [ None; Some Copy; Some Symlink; Some Hardlink ]
+  else
+    [ None; Some Symlink; Some Copy; Some Hardlink ]
```

Comment:
I'm a bit confused: where do we actually change the default order of preference depending on `Sys.win32`? I couldn't find this.

Reply:
In fact, it looks like we are doing simply `List.find_map` over all preferences, regardless of the operating system, so it looks like some logic is missing.

Reply:
Before this PR, on Win32 we would implicitly map `Symlink` to `Copy`. This list is used in `Build_config.set`. BTW, this change did fix the Windows CI: before I pushed the second commit, this is what happened with the first commit: https://github.com/ocaml/dune/runs/4399129979?check_suite_focus=true#step:10:37

Reply:
Oh, thanks, I found it! Strangely, it uses `Sandbox_mode.all` instead of `all_except_patch_back_source_tree`, which is why, I guess, I failed to find it. Should this be changed to `all_except_patch_back_source_tree`? Otherwise, if the user forbids all sensible modes, Dune is going to pick up the crazy `patch_back_source_tree` :)

Reply:
(Happy to change `Build_config.set` in a separate commit if you like.)

Reply:
Yes, absolutely. Good catch!